test/test_large_cohort.py

```diff
@@ -127,111 +141,297 @@ def _mock_cohort(dataset_id: str):
     return mc
 
 
-class TestAllLargeCohortMethods:
+def decompress_into_job_tmp(tmp_path: Path, compressed_paths: list[str]):
+    """
+    Takes a list of compressed paths, and decompresses them into a job temp location
+    """
+    for path in compressed_paths:
+        with ZipFile(path, 'r') as zip_ref:
+            zip_ref.extractall(tmp_path)
 
-    @pytest.mark.integration
-    def test_with_sample_data(self, mocker: MockFixture, tmp_path: Path):
-        """
-        Run entire workflow in a local mode.
-        """
-        conf = create_config(
-            tmp_path=tmp_path,
-            sequencing_type='genome',
-            gnomad_prefix=gnomad_prefix,
-            broad_prefix=broad_prefix,
-        )
-        set_config(
-            conf,
-            tmp_path / 'config.toml',
-            merge_with=[DEFAULT_CONFIG, LARGE_COHORT_CONFIG],
-        )
 
-        mocker.patch(
-            'cpg_workflows.inputs.deprecated_create_cohort',
-            lambda: _mock_cohort(conf.workflow.dataset),
-        )
-        # skip can_reuse, implicit skip of existence checks
-        mocker.patch('cpg_workflows.large_cohort.combiner.can_reuse', lambda x: False)
-        mocker.patch('cpg_workflows.large_cohort.relatedness.can_reuse', lambda x: False)
-        mocker.patch('cpg_workflows.large_cohort.site_only_vcf.can_reuse', lambda x: False)
+def test_combiner(mocker: MockFixture, tmp_path: Path):
 
-        test_cohort: MultiCohort = cpg_workflows.inputs.deprecated_create_cohort()
-        gvcf_paths: list[str] = [str(sg.gvcf) for sg in test_cohort.get_sequencing_groups()]
+    conf = create_config(tmp_path)
+    set_config(
+        conf,
+        tmp_path / 'config.toml',
+        merge_with=[DEFAULT_CONFIG, LARGE_COHORT_CONFIG],
+    )
+
+    mocker.patch(
+        'cpg_workflows.inputs.deprecated_create_cohort',
+        lambda: _mock_cohort(conf.workflow.dataset),
+    )
+    # skip can_reuse, implicit skip of existence checks
+    mocker.patch('cpg_workflows.large_cohort.combiner.can_reuse', lambda x: False)
 
+    test_cohort: MultiCohort = cpg_workflows.inputs.deprecated_create_cohort()
+    gvcf_paths: list[str] = [str(sg.gvcf) for sg in test_cohort.get_sequencing_groups()]
+
+    try:
         start_query_context()
+    except hl.utils.java.FatalError as e:
+        pass
```

Comment:
Shouldn't we die if hail throws a fatal error? Interested to hear the reasoning

Reply:
Hail doesn't tolerate multiple instances, but the cleaner approach here would be to drop this startup into a session-scoped conftest fixture. That would still allow detection of real failures

Reply:
While I remember and before I forget - this is an error when Hail (in Java) tries to start a second session. The FatalError Is just the first error we can catch in Python.